src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,843 @@
+First, the field \c page->modify->instantiated is set to \c true, and the \c page_del
+field is retained from the prior deleted state.
+Second, if the transaction that deleted the page had not resolved yet when the
+instantiation happened, the field \c page->modify->inst_updates contains an array of the
+\c WT_UPDATE structures used to do the instantiation.
+
+These two conditions are orthogonal and resolve independently; when both are resolved the
+page is thenceforth a completely normal in-memory page.
+
+Note that instantiated pages are not automatically marked dirty.
+(See the notes on this point in the @ref ft_instantiation section.)
+
+Also note that it is not possible to fast-truncate an instantiated page.
+Only on-disk pages can be fast-truncated.
+See @ref ft_truncate.
+
+@subsection ft_instantiated_instantiated Instantiated flag and \c page_del
+
+The \c instantiated flag and the \c page_del structure are retained for the benefit of
+parent internal page reconciliation.
+See @ref ft_internal_reconciliation below.
+
+When the page is itself reconciled, or if the transaction that deleted it rolls back, the
+\c instantiated flag is cleared and the \c page_del structure is discarded.
+
+Note that the \c page_del field of an instantiated page should not be used to make
+operational decisions.
+Additional updates might have been applied to the page since the instantiation happened;
+these may contradict or obsolete the deletion information.
+
+@subsection ft_instantiated_updates \c inst_updates
+
+Meanwhile, the \c inst_updates field is kept independently until the transaction that
+created it is resolved.
+It in effect belongs to that transaction and is neither needed nor used by anything else,
+with one exception: eviction checks whether it is non-NULL before evicting the page.
+(Pages with an instantiated but uncommitted truncate cannot be evicted.)
+
+Because it is possible for the page to split between instantiation and transaction
+resolution, finding the updates created during instantiation to resolve them is
+problematic.
+The \c inst_updates field makes this possible.
+
+Upon resolution (either commit or rollback) the \c inst_updates field is discarded.
+
+@section ft_instantiation Instantiation
+
+Instantiation is the process by which an on-disk deleted page (ref in state \c
+WT_REF_DELETED) is converted into an in-memory page (ref in state \c WT_REF_MEM) with all
+its items explicitly deleted with tombstone updates.
+<i>Semantically, this is an identity transformation.</i>
+
+This occurs under two sets of circumstances: first, if a thread that cannot yet see the
+deletion tries to read from the deleted page; and second, if a search lands on the deleted
+page's portion of the namespace.
+It can happen either before or after the transaction that deleted the page is resolved.
+(But not at the same time; locking the ref prevents that.)
+
+(Note that for searches that are positioning to do updates, the instantiation is
```

Comment:
This whole paragraph seems too big and useful to be between bracket, we can probably remove them :)

Reply:
Removed them in [f3a555c](https://github.com/wiredtiger/wiredtiger/pull/8237/commits/f3a555cfd7ddc3655258d577632287a485c5676a).

There is no ticket that I know of. It is on my private "would be nice sometime" list along with other expensive and impractical things. If you think it's worth filing a ticket on, I can do that...